cassdegrees/ui/views.py

```diff
@@ -95,3 +95,77 @@ def sampleform(request):
 
 def create_subplan(request):
     return render(request, 'createsubplan.html')
+
+
+# inspired by the samepleform function created by Daniel Jang
+def manage_courses(request):
+    # Reads the 'action' attribute from the url (i.e. manage/?action=Add) and determines the submission method
+    action = request.GET.get('action', 'Add')
+    if   action == 'Add'   : method = 'post'
+    elif action == 'Edit'  : method = 'patch'
+    elif action == 'Delete': method = 'delete'
+    else:
+         action =  'Add'   ; method = 'post'
+
+    action = 'Add'; method = 'post' # TODO Remove once other methods are implemented
+
+    courses = requests.get(request.build_absolute_uri('/api/model/course/?format=json')).json()
```

Comment:
Wait, is this request made even for POST requests, which don't appear to use 'courses' at all? This doesn't seem right.

Reply:
@JackSekoranja

Reply:
No longer relevant - main template is now always rendered.